### PR TITLE
daemon: Add md to daemon call

### PR DESCRIPTION
**Pull request by Joelgranados**
Include a md_dev argument for the daemon executable.

Signed-off-by: Joel Granados <j.granados@samsung.com>

### CODE REVIEW

File: src/flexalloc_mm.c [hunk 1]
```diff
@@ -1722,8 +1722,19 @@ fla_open(struct fla_open_opts *opts, struct flexalloc **fs)
     goto free_dev_uri;
   }
 
+  if(md_dev != dev)
+  {
+    (*fs)->dev.md_dev_uri = strdup(opts->md_dev_uri);
+    if ((*fs)->dev.md_dev_uri == NULL)
+    {
+      err = -ENOMEM;
+      goto free_dev_uri;
```
Comment (jwdevantier) [inline src/flexalloc_mm.c:1731]: If this fails the allocation at line 1718 (opts->dev_uri) needs to be freed also, so you'll need a new label above free_dev_uri to handle this case.

File: src/flexalloc_daemon_base.c [hunk 2]
```diff
@@ -510,14 +511,30 @@ fla_daemon_fs_init_rq(struct fla_daemon_client *client, int sock_fd)
...
+  if(md_dev_uri_len == 0)
+    client->flexalloc->dev.md_dev_uri = NULL;
+  else
+  {
+    read_ptr += sizeof(size_t);
+    client->flexalloc->dev.md_dev_uri = strndup(read_ptr, md_dev_uri_len);
+    if (!client->flexalloc->dev.md_dev_uri)
+    {
+      err = -ENOMEM;
+      return err;
```
Comment (jwdevantier) [inline src/flexalloc_daemon_base.c:534]: here you will have to clean up the string allocated @ 516 (`client->flexalloc->dev.dev_uri`)

File: src/flexalloc_daemon_base.c [hunk 3]
```diff
@@ -510,14 +511,30 @@ fla_daemon_fs_init_rq(struct fla_daemon_client *client, int sock_fd)
   client->flexalloc->dev.lb_nbytes = client->flexalloc->geo.lb_nbytes;
   read_ptr += sizeof(struct fla_geo);
 
-  read_ptr += sizeof(size_t); // ignore string length field
-  client->flexalloc->dev.dev_uri = strdup(read_ptr);
```
Comment (jwdevantier) [inline src/flexalloc_daemon_base.c:514]: I like this cleanup :)

**Review by jwdevantier — CHANGES_REQUESTED** [review]
left some comments where memory must be freed.

Additionally, you will also have to add `free(fs->dev.md_dev_uri)` at `fla_close_noflush` in `flexalloc_mm.c`

There might also similarly be the need to free the `md_dev_uri` field in the daemon code somewhere. You should double-check :)

**Comment by Joelgranados** [comment]
Addressed all @jwdevantier comments.

**Pull request merged**